Studio Blueprints: Add site - options modal

bcotrim opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "Fixes" keyword and use "Related to" instead.
-->

## Related issues

- Fixes STU-654

## Proposed Changes

- Add new options modal for Add site workflow
- Fixes feature flag hook, to correctly get the defaults and update the flag values

<img width="918" height="844" alt="image" src="https://github.com/user-attachments/assets/1425669b-c01e-4e6b-96af-070595c1d329" />

Note: none of the options are working, we will implement each one of the options in separate follow-up PRs

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

- `ENABLE_BLUEPRINTS=true npm start`
- Press "Add site"
- You should see the options modal as shared above
- Check if you can close the modal
- Disable blueprints flag from the Electron menu or restart Studio by running `npm start`
- Add site should work as previously

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Have you checked for TypeScript, React or other console errors?

bcotrim committed:
add site - options modal

ivan-ottinger reviewed (commented):
Great work, Bernardo! I am still reviewing the code, but noticed two things, so wanted to share them with you already:

1. The icon of the second option is different than in the design:

<img width="2536" height="1792" alt="Markup on 2025-07-31 at 10:10:14" src="https://github.com/user-attachments/assets/07eecb53-fa15-4acd-955f-f9dafa12f912" />

2. The window dragging is interfering with the modal closing button and it can be tricky to close the modal:

https://github.com/user-attachments/assets/7e7933de-1557-407e-8c98-8490f6cf6de7

ivan-ottinger reviewed (commented):
The modal renders nicely and I did not observe any regressions regarding the legacy modal. 👍🏼 

Apart from my previous comment, here's one more thing we could consider:

I wonder if it would make sense to use HStack, VStack, Text and Heading components from the WordPress core instead of divs - since we are now moving to these components in Calypso as well.

nightnei commented:
>Check if you can close the modal

@bcotrim  I am continuing reviewing the code, but regarding UI - the close button is partially clickable:
https://github.com/user-attachments/assets/40ca2c3c-9753-4de5-b20e-0bfe696ab90d

nightnei commented:
> I wonder if it would make sense to use HStack, VStack, Text and Heading components from the WordPress core instead of divs - since we are now moving to these components in Calypso as well.

IMO - Studio is independent app, we should not make any dependency on dotcom, even regarding some rules, styleguides, etc.
Also, even already Studio uses npm instaed of yarn, we don't follow the same state-management libs and rules, so in general - I would not look at Calypso. And I woudl continue using divs.

nightnei commented on src/components/add-site/index.tsx:
```diff
@@ -0,0 +1,76 @@
+		openModal();
+	} );
+
+	// If blueprints is disabled, use the existing component
```
I think code is self explanatory
```suggestion
```

nightnei commented on src/components/add-site/index.tsx:
```diff
@@ -0,0 +1,76 @@
+		);
+	}
+
+	// If blueprints is enabled, show modal with options
```
I think code is self explanatory
```suggestion
```

ivan-ottinger commented:
> IMO - Studio is independent app, we should not make any dependency on dotcom, even regarding some rules, styleguides, etc.

I have seen VStack and HStack being used it some places in Studio already so that what made me think we could consider it further. Mostly from the perspective of unifying design and simplifying styling as I could see how using those core components reduces the need for custom styles for example.

nightnei commented on src/components/modal.tsx:
```diff
@@ -3,7 +3,11 @@ import { useI18n } from '@wordpress/react-i18n';
-export default function Modal( { className, ...rest }: ComponentProps< typeof WPModal > ) {
+export default function Modal( {
+	className,
+	isFullScreen,
```
NIT Typically prefix `is` isn't used. For example `disabled`. And I think it makes sense, since this has better readability `<Modal fullScreen onRequestClose={ closeModal }>` instead of `<Modal isFullScreen onRequestClose={ closeModal }>`.

```suggestion
	fullScreen,
```

nightnei commented on src/hooks/use-feature-flags.tsx:
```diff
@@ -26,15 +31,44 @@ interface FeatureFlagsProviderProps {
 export const FeatureFlagsProvider: React.FC< FeatureFlagsProviderProps > = ( { children } ) => {
-	const [ featureFlags, setFeatureFlags ] = useState< FeatureFlagsContextType >( {
-		...defaultFeatureFlags,
+	const [ featureFlags, setFeatureFlags ] = useState< FeatureFlagsContextType >( () => {
```
Could you please elaborate on all changes inside this module?
I didn't get why do we need them, also they look quite complicated and also we didn't have such changes before so I wonder - can we avoid them as we did it before?

bcotrim replied:
We had a bug with how feature flags were being updated/preserved, so changing the value in the menu wasn't being reflected in the app. But you are right, that logic was complex, and we didn't need all that.

The simplified implementation now properly handles the three sources of feature flags (defaults → appGlobals → API) in a cleaner way, and it correctly preserves API flags when the menu triggers an update.

Let me know if it is clear now.

nightnei reviewed (changes requested):
Also, let's move and initially create folder for it inside `src/modules`, e.g. `src/modules/add-site/*`

bcotrim committed:
fullscreen modal

bcotrim commented:
@nightnei @ivan-ottinger thanks for the reviews!
I think I've addressed the feedback.

I decided to create a new modal for full-screen, the one from wordpress components isn't completely full-screen and my workaround was causing the strange behavior in the close button. Let me know what you think.

~As for the icon, I will follow-up with design and update in a follow-up PR if you agree.~

bcotrim committed:
feature flags logic

bcotrim committed:
adjust icons

nightnei commented on src/hooks/use-feature-flags.tsx:
```diff
@@ -26,15 +26,24 @@ interface FeatureFlagsProviderProps {
-	const [ featureFlags, setFeatureFlags ] = useState< FeatureFlagsContextType >( {
-		...defaultFeatureFlags,
+	const [ featureFlags, setFeatureFlags ] = useState< FeatureFlagsContextType >( () => {
+		if ( window.appGlobals ) {
+			return {
+				...defaultFeatureFlags,
+				...window.appGlobals,
+			};
+		}
+		return { ...defaultFeatureFlags };
```
Hm, can't we go this way? it should work:
<img width="253" height="91" alt="Screenshot 2025-07-31 at 18 53 48" src="https://github.com/user-attachments/assets/aa252c17-0284-4a2b-ae5b-3371b7e815cc" />

```suggestion
		return {
			...defaultFeatureFlags,
			...window.appGlobals,
		};
```

nightnei reviewed (approved):
Looks great 👍
Left only one small comment

bcotrim committed:
address pr feedback

ivan-ottinger commented on src/modules/add-site/components/add-site-options.tsx:
```diff
@@ -0,0 +1,85 @@
+					{ description }
+				</Text>
+			</VStack>
+			<Icon className="mt-0.5" icon={ chevronRight } size={ 24 } fill="#949494" />
```
We may want to use `chevronLeft` for RTL as currently it points the same as in LTR:

<img width="2644" height="2268" alt="Markup on 2025-08-01 at 09:35:31" src="https://github.com/user-attachments/assets/6fca0288-1f7c-497f-ab3c-b8b0627135df" />

Also, I think the title and description should be starting on the right side when RTL locale is set.

ivan-ottinger reviewed (commented):
Thank you for making the updates, Bernardo!

I have noticed one more thing: each icon has a different size (36px vs 32px vs 24px).

It is mostly visible with the "Start from blueprint" button where the title and description is then shifted to the side a bit:

<img width="2568" height="1532" alt="Markup on 2025-08-01 at 09:25:37" src="https://github.com/user-attachments/assets/d9cd9867-be7e-49fb-88b2-bf9125950157" />

---

I left one more comment related to RTL below.

bcotrim committed:
rtl adjustments

ivan-ottinger reviewed (approved):
Nice! Icons are now the same size and both LTR and RTL are rendered correctly!

| LTR | RTL |
|--------|--------|
| <img width="2580" height="2192" alt="Markup on 2025-08-01 at 11:32:14" src="https://github.com/user-attachments/assets/51a2970a-78ee-43bc-a3e7-61aca934ce05" /> | <img width="2528" height="2156" alt="Markup on 2025-08-01 at 11:32:44" src="https://github.com/user-attachments/assets/71f136d1-f7a4-497a-8e01-8a158af4b92b" /> |
